x-pack/legacy/plugins/siem/public/components/ml_popover/api.tsx

```diff
@@ -32,10 +43,16 @@ export const groupsData = async (headers: Record<string, string | undefined>): P
         ...headers,
       },
     });
+    const message = await parseError(response);
+    if (message != null) {
```

Comment:
if you throw an error in the parser, the catch will do the same job than the if

Reply:
Updated each area and was able to reduce the boiler plate.